s3.py

```diff
@@ -35,34 +38,67 @@ def feed_url(cls, filename, extension='.xml', open_access=True):
             filename += extension
         return root + filename
 
-    def delete_batch(self, keys, _db=None):
+    def delete_batch(self, keys, _db=None, external_hosts=None):
         """Deletes files identified by their keys (i.e. mirror urls)
         from s3 bucket and--if a database session is provided--their
         saved Representations.
 
         This method is intended for utility use.
+
+            :param _db: A database session to delete Representations,
+                Resources, and Hyperlinks associated with a particular
+                S3 key / Representation.mirror_url.
+            :param external_hosts: A list of strings representing hosts
+                that are not relevant S3 buckets.
         """
         requests = list()
+        keys = [unicode(k) for k in keys]
+
         if _db:
-            from model import Representation, get_one
-            for key in keys:
-                representation = get_one(_db, Representation, mirror_url=unicode(key))
-                if representation:
-                    logging.info("DELETED Representation for %s" % key)
-                    _db.delete(representation)
-            _db.commit()
+            from core.model import Representation, Resource
+            representations = _db.query(Representation)\
+                .join(Representation.resource).join(Resource.links)\
+                .options(
+                    joinedload(Representation.resource)\
+                    .joinedload(Resource.links))\
+                .filter(Representation.mirror_url.in_(keys))
+
+            for representation in representations:
+                url = representation.mirror_url
+                resource = representation.resource
+                logging.info('Deleting %s...', url)
+                if resource:
+                    count = len(resource.links)
+                    [_db.delete(link) for link in resource.links]
+                    logging.info("\t- DELETED %d Hyperlink(s)", count)
+
+                    _db.delete(resource)
+                    logging.info("\t- DELETED Resource %r", resource)
+
+                _db.delete(representation)
+                logging.info("\t- DELETED Representation %r", representation)
```

Comment:
It's probably not a problem because the delete is happening all at once when its committed (outside the method), but it's a good point. I'm gonna switch it so it better represents the data relationship.